src/adapter/aws-lambda/handler.ts

```diff
@@ -24,41 +39,94 @@ interface APIGatewayProxyResult {
   isBase64Encoded: boolean
 }
 
+/**
+ * Accepts events from API Gateway/ELB(`APIGatewayProxyEvent`) and directly through Function Url(`APIGatewayProxyEventV2`)
+ */
 export const handle = (app: Hono) => {
-  return async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
+  return async (
+    event: APIGatewayProxyEvent | APIGatewayProxyEventV2
+  ): Promise<APIGatewayProxyResult> => {
     const req = createRequest(event)
     const res = await app.fetch(req)
-    const arrayBuffer = await res.arrayBuffer()
-    const result: APIGatewayProxyResult = {
-      statusCode: res.status,
-      body: String.fromCharCode(...new Uint8Array(arrayBuffer)),
-      headers: {},
-      isBase64Encoded: false,
-    }
-
-    res.headers.forEach((value, key) => {
-      result.headers[key] = value
-    })
-
-    return result
+
+    return createResult(res)
+  }
+}
+
+async function createResult(res: Response): Promise<APIGatewayProxyResult> {
+  const isBase64Encoded = res.body instanceof ReadableStream
+  const body = isBase64Encoded ? await fromReadableToString(res) : await fromBufferTostring(res)
+
+  const result: APIGatewayProxyResult = {
+    body,
+    headers: {},
+    statusCode: res.status,
+    isBase64Encoded,
   }
+
+  res.headers.forEach((value, key) => {
+    result.headers[key] = value
+  })
+
+  return result
 }
 
-const createRequest = (event: APIGatewayEvent) => {
-  const url = `https://${event.requestContext.domainName}${event.path}`
+function createRequest(event: APIGatewayProxyEvent | APIGatewayProxyEventV2) {
+  const queryString = extractQueryString(event)
+  const urlPath = isProxyEventV2(event)
+    ? `https://${event.requestContext.domainName}${event.rawPath}`
+    : `https://${event.requestContext.domainName}${event.path}`
+
+  const url = queryString ? `${urlPath}?${queryString}` : urlPath
+
   const headers = new Headers()
   for (const [k, v] of Object.entries(event.headers)) {
     if (v) headers.set(k, v)
   }
-  const method = event.httpMethod
 
   const requestInit: RequestInit = {
-    headers: headers,
-    method: method,
+    headers,
+    method: event.httpMethod,
   }
 
   if (event.body) {
-    requestInit.body = event.isBase64Encoded ? atob(event.body) : event.body
+    requestInit.body = event.isBase64Encoded ? Buffer.from(event.body, 'base64') : event.body
```

Comment:
Use `atob`/`btoa` instead of `Buffer`. 

We can use `Buffer` on AWS Lambda with Node.js, but the Hono core consistently uses `btoa`/`atob` instead of `Buffer`, so use that.